### ISSUE TITLE
Enhance Gradle task customisation to support requirements of projects using Kotlin

### ISSUE THREAD

**Issue by wilkinsona** [wilkinsona/project-generator#9]
A project that uses Kotlin needs to be able to do task customisation like this:

```
compileKotlin {
    kotlinOptions {
        freeCompilerArgs = ["-Xjsr305=strict"]
        jvmTarget = "1.8"
    }
}
```

The current implementation was driven by requirements from Spring REST Docs and only supports customisation like this:

```
asciidoctor { 
    inputs.dir snippetsDir 
    dependsOn test 
}
```

We need to support the nesting (`kotlinOptions` in this specific case). We also need to look at the difference between `jvmTarget = "1.8"` and `jvmTarget "1.8"`. We may need to support both.

**Comment by wilkinsona** [comment]
The `compileKotlin` customization can be written like this:

```
compileKotlin {
	kotlinOptions.freeCompilerArgs = ["-Xjsr305=strict"]
	kotlinOptions.jvmTarget = "1.8"
}
```

We definitely need to support assignment.

**Comment by wilkinsona** [comment]
Fixed in 094b58cd3d5b4a2da47b1f3ef53043603347f141.
